src/session_impl.cpp

```diff
@@ -4567,6 +4567,8 @@ namespace {
 		std::vector<torrent*>& state_updates
 			= m_torrent_lists[aux::session_impl::torrent_state_updates];
 
+		if (state_updates.empty()) return;
```

Comment:
if the user asks for an update, and doesn't get one, the invariant for this call becomes impossibly difficult to use.

should the client expect a response? there's no way of knowing, so should the client have some time-out? if it hasn't received a response in 3 seconds, assume there has been no updates. That has several problems. (1) update frequency may be difficult to keep high, as you may want to time-out your first request before issuing another (2) it's wrong, since on some devices under certain load it may take more than 3 seconds for a response to come back